ingenialink/canopen/network.py

```diff
@@ -239,6 +244,42 @@ def scan_slaves(self) -> List[int]:
 
         return nodes  # type: ignore [no-any-return]
 
+    def scan_slaves_info(self) -> OrderedDict[int, SlaveInfo]:
+        """Scans for nodes in the network and return an ordered dict with the slave information.
+
+        Returns:
+            Ordered dict with the slave information.
+
+        """
+        slave_info: OrderedDict[int, SlaveInfo] = OrderedDict()
+        try:
+            slaves = self.scan_slaves()
+        except ILError:
+            return slave_info
+
+        if self._connection is None:
+            try:
+                self._setup_connection()
+            except ILError:
+                self._teardown_connection()
+                return slave_info
+
+        if self._connection is None:
+            return slave_info
+
+        for slave_id in slaves:
+            node = self._connection.add_node(slave_id)
+            product_code = convert_bytes_to_dtype(
+                node.sdo.upload(self.DRIVE_INFO_INDEX, self.PRODUCT_CODE_SUB_IX), REG_DTYPE.U32
+            )
+            revision_number = convert_bytes_to_dtype(
+                node.sdo.upload(self.DRIVE_INFO_INDEX, self.REVISION_NUMBER_SUB_IX), REG_DTYPE.U32
+            )
+            slave_info[slave_id] = SlaveInfo(int(product_code), int(revision_number))
+
+        self._teardown_connection()
```

Comment:
What happens if you are already connected to a drive and scan? You lose the connection, right?

Reply:
Done [cbc4306](https://github.com/ingeniamc/ingenialink-python/pull/377/commits/cbc430614679d6f2a8bc1caf8807e8f0369c2684)